### PR TITLE
Displaying item counts at Communities and Collections list view.

### LINKED ISSUE #1787: Add option to display Item counts for Communities and Collections
**Is your feature request related to a problem? Please describe.**
Please replicate in DSpace 7 the DSpace 6 XMLUI option to display items counts for Communities and Collections.

**Describe the solution you'd like**
The display in DSpace 6 XMLU is fine.
DSpace6_XMLUI_item_counts
DSpace6_XMLUI_item_counts2

**Describe alternatives or workarounds you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by damian-joz**
## References
* Fixes #1787
* Requires DSpace/DSpace#8843

## Description
Displaying item counts at Communities  views. It uses a new field from Collection and Community models. 

## Instructions for Reviewers
List of changes in this PR:
* Extending Collection and Community models
* Displaying items counts at /community-list and /communities views

To test it you should create at least one item in collection. Expected behaviour is to show item counts for communities and collections:

comm

## Checklist
_This checklist provides a reminder of what we are going to look for when reviewing your PR. You need not complete this checklist prior to creating your PR (draft PRs are always welcome). If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_

- [x] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & specs/tests), or I have provided reasons as to why that's not possible.
- [x] My PR passes [ESLint](https://eslint.org/) validation using `yarn lint`
- [x] My PR doesn't introduce circular dependencies (verified via `yarn check-circ-deps`)
- [ ] My PR includes [TypeDoc](https://typedoc.org/) comments for _all new (or modified) public methods and classes_. It also includes TypeDoc for large or complex private methods.
- [x] My PR passes all specs/tests and includes new/updated specs or tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [ ] If my PR includes new libraries/dependencies (in `package.json`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.
- [ ] If my PR includes new features or configurations, I've provided basic technical documentation in the PR itself.
- [x] If my PR fixes an issue ticket, I've [linked them together](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).

### CODE REVIEW

**Review by kshepherd — APPROVED** [review]
+1, tested this along with https://github.com/DSpace/DSpace/pull/8843 and it works as advertised. Great to see the updated spec tests. The imports could use a space between the braces and the class eg `{ excludeFromEquals }` instead of `{excludeFromEquals}`  but I don't think that's enforced or consistent across dspace-angular anyway, just a tiny nitpick you could fix along with the main branch rebase+push ;)

(if you use IntelliJ IDEA, there is a code style setting somewhere to tell it to add that whitespace when it generates imports, not sure about other IDEs)

thanks @damian-joz !

**Comment by kshepherd** [comment]
@damian-joz one other note - the counts appear in community and collection list pages/comps but not beside community names in the home page community list. I wasn't sure if that was intended or not

**Comment by damian-joz** [comment]
> one other note - the counts appear in community and collection list pages/comps but not beside community names in the home page community list. I wasn't sure if that was intended or not

@kshepherd  Thanks for the review. That was intended because I am not sure how it works in DSpace 6.x. Adding this wan't be a problem. Just let me know.

**Review by tdonohue — COMMENTED** [review]
@damian-joz : Overall this looks good.  I did notice though that these item counts *only* appear on the Community List (`/community-list`) and are missing from the homepage (where communities are also listed).  That's fine for now, but I wanted to call it out in case you meant for it to also display on the homepage.

One other comment inline below.

File: src/app/community-list-page/community-list/community-list.component.html [hunk 1]
```diff
@@ -37,6 +37,7 @@ <h5 class="align-middle pt-2">
         <a [routerLink]="node.route" class="lead">
           {{node.name}}
         </a>
+        <span *ngIf="node.payload.archivedItemsCount" class="archived-items-lead">[{{node.payload.archivedItemsCount}}]</span>
```
[the remark this answers is not in the document]
Comment (damian-joz) [reply src/app/community-list-page/community-list/community-list.component.html]: Done.

**Review by tdonohue — APPROVED** [review]
👍 Thanks again @damian-joz !  It works well with the counts both enabled and disabled. Code and tests look good as well.

**Comment by tdonohue** [comment]
Merging for 7.6 as this has had +2 approval and I've verified that all my feedback & @kshepherd 's feedback has been addressed.

**Comment by alanorth** [comment]
Re-posting my comment from the backend part of this PR that was meant to be here.

This works well. Thank you @damian-joz. I'm wondering why the item counts use Bootstrap `lead` classes, though? This increases the font size and makes them stand out too much in my opinion.

This is my custom theme, but the effect is the same in the default theme:

<p align="center">
  
</p>

In DSpace 6 XMLUI the item counts were the same color as the default body text. If we want to have them look subtly different, we could use [Bootstrap's `font-weight-light`](https://getbootstrap.com/docs/4.6/utilities/text/#font-weight-and-italics) class:

<p align="center">
  
</p>

**Pull request merged**